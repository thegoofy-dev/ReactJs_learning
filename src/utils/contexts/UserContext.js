import { createContext } from "react";

export const UserContext = createContext({
    id:0, 
    username: "Guest",
    email: "default@gmail.com",
    displayName: "Welcome Guest",
});