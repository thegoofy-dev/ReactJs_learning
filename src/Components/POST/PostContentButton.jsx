import { useContext, useState } from "react";
import { UserContext } from "../../utils/contexts/UserContext";

export function PostContentButton() {
  const { username, setData } = useContext(UserContext);
  const [value, setValue] = useState("");

  return (
    <div>
      <hr />
      <div>
        <h3>Post Content Button</h3>
      </div>
      <span>UserName: {username}</span> <br />
      <label htmlFor="updateName">UpdateName:</label>
      <input
        id="updateName"
        name="updateName"
        type="text"
        value={value}
        onChange={(e) => setValue(e.target.value)}
      />
      <button
        onClick={() => {
          setData((currentState) => ({
            ...currentState,
            name: value,
          }));
        }}
      >
        Save Display Name
      </button>
    </div>
  );
}
