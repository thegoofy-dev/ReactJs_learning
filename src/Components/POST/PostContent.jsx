import { useContext } from "react";
import { PostContentButton } from "./PostContentButton";
import { UserContext } from "../../utils/contexts/UserContext";

export function PostContent() {
  const userContextData = useContext(UserContext);

  return (
    <div>
      <hr />
      <div>
        <h3>Post Content</h3>
      </div>
      <PostContentButton />
      <span>Email: {userContextData.email}</span>
    </div>
  );
}
