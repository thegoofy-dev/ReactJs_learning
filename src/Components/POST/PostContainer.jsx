import { UserContext } from "../../utils/contexts/UserContext";
import { PostContent } from "./PostContent";
import { useContext } from "react";

export function PostContainer() {
  const userContextData = useContext(UserContext);

  return (
    <div>
      <hr />
      <div>
        <h3>Post Container</h3>
      </div>
      <PostContent />
      <div>DisplayName: {userContextData.name}</div>
      <div>ID: {userContextData.id}</div>
      <div>Email: {userContextData.email}</div>
      <div>UserName: {userContextData.username}</div>
    </div>
  );
}
