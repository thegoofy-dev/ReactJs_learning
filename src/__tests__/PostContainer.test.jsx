import { describe, it, expect } from "vitest";
import { screen, render } from "@testing-library/react";
import { PostContainer } from "../Components/POST/PostContainer";
import { UserContext } from "../utils/contexts/UserContext";

describe("Display Context values", () => {
  const mockUserContextData = {
    id: 101,
    username: "testUser",
    email: "testEmail",
    user: "test",
    setData: () => {},
  };

  it("should match the snapshot", () => {
    const container = render(
      <UserContext.Provider value={mockUserContextData}>
        <PostContainer />
      </UserContext.Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it("should display username", () => {
    render(
      <UserContext.Provider value={mockUserContextData}>
        <PostContainer />
      </UserContext.Provider>
    );

    const elements = screen.getAllByText("UserName: testUser");
    expect(elements.length).toBeGreaterThan(0);
    elements.forEach((element) => expect(element).toBeInTheDocument());
  });
});
