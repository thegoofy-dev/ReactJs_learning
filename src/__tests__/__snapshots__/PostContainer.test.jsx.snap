// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Display Context values > should match the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <hr />
        <div>
          <h3>
            Post Container
          </h3>
        </div>
        <div>
          <hr />
          <div>
            <h3>
              Post Content
            </h3>
          </div>
          <div>
            <hr />
            <div>
              <h3>
                Post Content Button
              </h3>
            </div>
            <span>
              UserName: 
              testUser
            </span>
             
            <br />
            <input
              type="text"
              value=""
            />
            <button>
              Save Display Name
            </button>
          </div>
          <span>
            Email: 
            testEmail
          </span>
        </div>
        <div>
          DisplayName: 
        </div>
        <div>
          ID: 
          101
        </div>
        <div>
          Email: 
          testEmail
        </div>
        <div>
          UserName: 
          testUser
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <hr />
      <div>
        <h3>
          Post Container
        </h3>
      </div>
      <div>
        <hr />
        <div>
          <h3>
            Post Content
          </h3>
        </div>
        <div>
          <hr />
          <div>
            <h3>
              Post Content Button
            </h3>
          </div>
          <span>
            UserName: 
            testUser
          </span>
           
          <br />
          <input
            type="text"
            value=""
          />
          <button>
            Save Display Name
          </button>
        </div>
        <span>
          Email: 
          testEmail
        </span>
      </div>
      <div>
        DisplayName: 
      </div>
      <div>
        ID: 
        101
      </div>
      <div>
        Email: 
        testEmail
      </div>
      <div>
        UserName: 
        testUser
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
